specification.md

```diff
@@ -1631,6 +1631,168 @@ ThrottleSubmitter().start()
 ```
 
 
+#### Submit a malformed or unsatisfiable job
+
+Use the exception types to distinguish between re-triable or not
+
+```python
+import jpsi
+
+jex = jpsi.JobExector()
+job_1 = jpsi.Job()
+job_2 = jpsi.Job()
+
+spec_1 = jpsi.JobSpec()
+spec_2 = jpsi.JobSpec()
+
+spec_1.executable = '/bin/true'
+spec_2.executable = True   # type error
+
+job_1.specification = spec_1
+job_2.specification = spec_2
+
+try:
+    jex.submit(job_1)
+except jpsi.InvalidJobException:
+    # this should not happen
+    assert(True)
+except jpsi.SubmitException:
+    # this *can* happen, dependent on backend state and policies
+    print('could not submit job - try again later')
+
+try:
+    jex.submit(job_2)
+except jpsi.InvalidJobException as e:
+    print('submission failed: %s' % e)
+else:
+    assert(True)  # the above should have raised an `InvalidJobException`
+
+job_1.wait()
+```
+
+
+#### Run a job with P total processes where each process gets C cpus and G gpus
+
+```python
+import jpsi
+
+jex = jpsi.JobExector()
+job = jpsi.Job()
+
+P = 10
+C = 4
+G = 1
+
+res_spec = jpsi.ResourceSpecification({
+              'process_count'    : P,
+              'cores_per_process': C,
+              'gpus_per_process' : G,
+           })
+job_spec = jpsi.JobSpecification({
+              'executable': 'workload.py',
```

Comment:
```suggestion
              'executable': 'echo',
```

Assuming `workload.py` isn't provided somewhere else, this example will error out if someone blindly copies and pastes it.  Maybe we want to sacrifice some "realism" to increase the probability that it "just works"?  Just a though.  Feel free to ignore.